View of top page

THitokuse opened the pull request:
WHAT: トップページの作成をした。
WHY: 一番最初に表示される画面のため。

TaichiroHasegawa committed:
haml導入 gemfileにhamlを導入

gem 'haml-rails'
gem 'erb2haml'
gem 'font-awesome-rails'

TaichiroHasegawa committed:
cssをscssとして導入

application.cssをapplication.scssに変更
@import "font-awesome";
@import "reset";
をインポート

reset.scssを作成

TaichiroHasegawa committed:
トップページのheader作成

index.html.haml、header.scssによるheaderを作成
application.scssでheaderをインポート

TaichiroHasegawa committed:
トップページのfooterを作成

index画面、footer.scssの作成
application.scssにfooter.scssをインポート

TaichiroHasegawa committed:
トップページにスライドの1ページ目を付与

index画面、slide.scssを追加
application.scssにslide.scssをインポート

TaichiroHasegawa committed:
トップページのitem一覧部分表示

index画面、item.scssを追加
application.scssにitem.scssをインポート

TaichiroHasegawa committed:
トップページに出品ボタンを作成

index画面、sell_btn.scssを追加
application.scssにsell_btn.scssをインポート

TaichiroHasegawa committed:
不要なファイルを削除

mailer.html.erb
mailer.text.erb
を削除

TaichiroHasegawa committed:
rootにitems#index画面を設定

トップページをitem#index画面に設定

TaichiroHasegawa committed:
application.htmlをhaml化

TaichiroHasegawa committed:
itemモデルを作成

name string型
price int型
prefecture_code int型
content text型
カラムを作成

TaichiroHasegawa committed:
itemimageモデルを作成

item_id
image string型
カラムを作成

itemとitemimageにアソシエーションを作成

TaichiroHasegawa committed:
image_uploaderを作り直すため削除

TaichiroHasegawa committed:
トップページにアイテムを表示

one_itemの部分テンプレート作成
index画面からone_itemの部分テンプレートの呼び出し
one_items.scssの作成
application.scssでone_items.scssのインポート

TaichiroHasegawa committed:
carrierwaveの導入 gemfileの追加

TaichiroHasegawa committed:
image_uploaderの追加

itemimageにimage_uploaderを付与

TaichiroHasegawa committed:
gemfile.lockのコンフリクト解消

TaichiroHasegawa committed:
gemfile.lockのコンフリクト解消

TaichiroHasegawa committed:
routingのコンフリクト解消

THitokuse committed:
Merge branch 'master' into view_of_top_page

exp-drill commented on test/models/itemimage_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
minitestで実装しますか？不要であれば削除お願いします！

THitokuse replied:
削除しました。

exp-drill commented on test/models/item_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
minitestで実装しますか？不要であれば削除お願いします！

THitokuse replied:
削除しました。

exp-drill commented on test/fixtures/items.yml:
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
minitestで実装しますか？不要であれば削除お願いします！

THitokuse replied:
削除しました。

exp-drill commented on test/fixtures/itemimages.yml:
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
minitestで実装しますか？不要であれば削除お願いします！

THitokuse replied:
削除しました。

exp-drill commented on test/controllers/items_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
minitestで実装しますか？不要であれば削除お願いします！

THitokuse replied:
削除しました。

exp-drill commented on config/routes.rb:
```diff
@@ -1,4 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get "items" => "items#index"
```
resourcesで実装しましょう！

THitokuse replied:
resourcesで実装しました。

exp-drill commented on app/views/shared/_one_items.html.haml:
```diff
@@ -0,0 +1,14 @@
+%section.items-box
+  = link_to do
+    %figure.items-box-photo
+      / -if item.itemimages.first.photo_url
```
後で使うことが想定されていますでしょうか？

THitokuse replied:
プルリク前にgithubなどで得たものをいろいろ試行錯誤した際に残ってしまったものでした。
削除しておきます。

exp-drill reviewed (commented):
ハッシュロケットは削除しましょう！

TaichiroHasegawa committed:
不要なtestコード削除

現在はテストコードを使わないため削除

TaichiroHasegawa committed:
不要なゴミ削除

プルリク前にgithubなどで得たものをいろいろ試行錯誤した際に残ってしまったものでした。

/ -if item.itemimages.first.photo_url
削除

TaichiroHasegawa committed:
items#indexをresourcesで記述

resources :items, only: :index
に変更

TaichiroHasegawa committed:
ハッシュロケット削除

ハッシュロケットを全て変更

TaichiroHasegawa committed:
ハッシュロケット削除

追加でハッシュロケットを削除

TaichiroHasegawa committed:
タイトル作成

メルカリ スマホで簡単 フリマアプリ
生成

TaichiroHasegawa committed:
ファビコン付与

https://www.recycle-tsushin.com/news/aa84d5e628153468ff17ca2b64dcd46febfdd9a0.png
ファビコンを導入しました。

TaichiroHasegawa committed:
不要なゴミの削除

不必要な部分の削除

exp-drill commented on app/assets/stylesheets/footer.scss:
```diff
@@ -0,0 +1,195 @@
+//colors
+$white: #fff;
```
これは変数用のファイル作成してimportした方がどのファイルにも使用できるので良いと思います！

THitokuse replied:
color.scssを作成してimportさせました。

TaichiroHasegawa committed:
color.scss導入

colorに変数定義
以下4項目
$white: #fff;
$black: #333;
$light_blue: #b3d4fc;
$black_40: rgba(0,0,0,0.4);

TaichiroHasegawa committed:
footerの色の変数化

TaichiroHasegawa committed:
headerの色の変数化

TaichiroHasegawa committed:
items.scssの色の変数化

TaichiroHasegawa committed:
one_items.scssの色の変数化

TaichiroHasegawa committed:
sell_btn.scssの変数化
